Dataset worked two days ago, now OpenCV error when loading into YOLO for training.

### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.


### Ultralytics YOLO Component

Train

### Bug

Two days ago I was working on training on two datasets, one with YOLO-seg and the other with YOLO-pose (both v11). My data is (and has always been) 8-bit grayscale images. I've succeeded in training on Google Colab before and have some decent results using the same datasets but wanted to retrain with some new training parameters. I got an error in the data augmentation suggesting that img.shape should have more than two dimensions ('img.shape[2] tuple index is out of bounds' or something along those lines). After some troubleshooting I turned my dataset into RGB images instead and that got me past the augmentation error. 

However, a new error occurs which suggests some images are being read as empty or not being read correctly:

`--> 237                     im = cv2.resize(im, (w, h), interpolation=cv2.INTER_LINEAR)
    238             elif not (h0 == w0 == self.imgsz):  # resize by stretching image to square imgsz
    239                 im = cv2.resize(im, (self.imgsz, self.imgsz), interpolation=cv2.INTER_LINEAR)

error: OpenCV(4.11.0) /io/opencv/modules/imgproc/src/resize.cpp:3845: error: (-215:Assertion failed) !dsize.empty() in function 'resize'`

I looked into this error in the general context of OpenCV and most solutions suggest that OpenCV may fail to read in an image if "/" are missing from file paths (I did not have this issue) or if there are blank spaces in a list of images or corrupt images that OpenCV cannot otherwise open. All of the images in my dataset are valid - and **MOST IMPORTANTLY** did not have either issue until yesterday. Does anyone know if OpenCV has updated in the past two days in some way that has created a bug with YOLO image augmentation and resizing? Having to turn my image data into RGB rather than 8-bit grayscale is not a big problem, but I'm lost for a solution on the current issue in the resize function. Any help finding a solution is appreciated!

### Environment

Ultralytics 8.3.114 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA L4, 22693MiB)
Setup complete ✅ (12 CPUs, 53.0 GB RAM, 41.0/235.7 GB disk)

Environment: Google Colab
Python version: 3.10

### Minimal Reproducible Example

results = model.train(data='/content/drive/MyDrive/YOLO_Pose/poses.yaml', epochs=600, imgsz= 640, batch=0.65, cache=True, workers=6, multi_scale=True, cos_lr=True, dfl=1.5, cls=0.75, box=7.75, degrees=90.0, shear=90.0, scale=0.9, mixup=0.15, copy_paste=0.4, flipud=0.25, fliplr=0.25, patience=200)

**I have also tried running training with all default settings and get the same error.**

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

`Ultralytics 8.3.114 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA L4, 22693MiB)
engine/trainer: task=pose, mode=train, model=yolo11m-pose.yaml, data=/content/drive/MyDrive/YOLO_Pose/poses.yaml, epochs=600, time=None, patience=200, batch=0.65, imgsz=640, save=True, save_period=-1, cache=True, device=None, workers=6, project=None, name=train2, exist_ok=False, pretrained=yolo11m-pose.pt, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=True, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.75, cls=0.75, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=90.0, translate=0.1, scale=0.9, shear=90.0, perspective=0.0, flipud=0.25, fliplr=0.25, bgr=0.0, mosaic=1.0, mixup=0.15, copy_paste=0.4, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, cfg=None, tracker=botsort.yaml, save_dir=runs/pose/train2

                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]     
  3                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]     
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  6                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  8                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1    990976  ultralytics.nn.modules.block.C2PSA           [512, 512, 1]                 
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 13                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 16                  -1  1    542720  ultralytics.nn.modules.block.C3k2            [1024, 256, 1, True]          
 17                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 19                  -1  1   1511424  ultralytics.nn.modules.block.C3k2            [768, 512, 1, True]           
 20                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 22                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 23        [16, 19, 22]  1   2261605  ultralytics.nn.modules.head.Pose             [1, [3, 2], [256, 512, 512]]  
YOLO11m-pose summary: 246 layers, 20,903,589 parameters, 20,903,573 gradients, 71.9 GFLOPs

Transferred 691/691 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.2±0.1 ms, read: 241.2±160.9 MB/s, size: 737.8 KB)
train: Scanning /content/drive/MyDrive/YOLO_Pose/labels/train... 974 images, 67 backgrounds, 0 corrupt: 100%|██████████| 1041/1041 [00:05<00:00, 201.86it/s]
train: New cache created: /content/drive/MyDrive/YOLO_Pose/labels/train.cache
WARNING ⚠️ cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.
  0%|          | 0/1041 [00:00<?, ?it/s]
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-12-ebcd4b24fc75> in <cell line: 0>()
----> 1 results = model.train(data='/content/drive/MyDrive/YOLO_Pose/poses.yaml', epochs=600, imgsz= 640, batch=0.65, cache=True, workers=6, multi_scale=True, cos_lr=True, dfl=1.5, cls=0.75, box=7.75, degrees=90.0, shear=90.0, scale=0.9, mixup=0.15, copy_paste=0.4, flipud=0.25, fliplr=0.25, patience=200)

13 frames
/usr/local/lib/python3.11/dist-packages/ultralytics/data/base.py in load_image(self, i, rect_mode)
    235                 if r != 1:  # if sizes are not equal
    236                     w, h = (min(math.ceil(w0 * r), self.imgsz), min(math.ceil(h0 * r), self.imgsz))
--> 237                     im = cv2.resize(im, (w, h), interpolation=cv2.INTER_LINEAR)
    238             elif not (h0 == w0 == self.imgsz):  # resize by stretching image to square imgsz
    239                 im = cv2.resize(im, (self.imgsz, self.imgsz), interpolation=cv2.INTER_LINEAR)

error: OpenCV(4.11.0) /io/opencv/modules/imgproc/src/resize.cpp:3845: error: (-215:Assertion failed) !dsize.empty() in function 'resize'`

Tried. Same error occurs, but now it happens as YOLO tries to load data for the first training epoch:

Ultralytics 8.3.114 🚀 Python-3.11.12 torch-2.6.0+cu124 CUDA:0 (NVIDIA L4, 22693MiB)
engine/trainer: task=pose, mode=train, model=yolo11m-pose.yaml, data=/content/drive/MyDrive/YOLO_Pose/poses.yaml, epochs=600, time=None, patience=200, batch=0.65, imgsz=640, save=True, save_period=-1, cache=False, device=None, workers=6, project=None, name=train, exist_ok=False, pretrained=yolo11m-pose.pt, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=True, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=True, opset=None, workspace=None, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.75, cls=0.75, dfl=1.5, pose=12.0, kobj=1.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=90.0, translate=0.1, scale=0.9, shear=90.0, perspective=0.0, flipud=0.25, fliplr=0.25, bgr=0.0, mosaic=1.0, mixup=0.15, copy_paste=0.4, copy_paste_mode=flip, auto_augment=randaugment, erasing=0.4, cfg=None, tracker=botsort.yaml, save_dir=runs/pose/train
Downloading https://ultralytics.com/assets/Arial.ttf to '/root/.config/Ultralytics/Arial.ttf'...
100%|██████████| 755k/755k [00:00<00:00, 141MB/s]Overriding model.yaml kpt_shape=[17, 3] with kpt_shape=[3, 2]
Overriding model.yaml nc=80 with nc=1

                   from  n    params  module                                       arguments                     
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]                 
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]     
  3                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]     
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  6                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
  8                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1    990976  ultralytics.nn.modules.block.C2PSA           [512, 512, 1]                 
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 13                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 16                  -1  1    542720  ultralytics.nn.modules.block.C3k2            [1024, 256, 1, True]          
 17                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           

 19                  -1  1   1511424  ultralytics.nn.modules.block.C3k2            [768, 512, 1, True]           
 20                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]              
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 22                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]          
 23        [16, 19, 22]  1   2261605  ultralytics.nn.modules.head.Pose             [1, [3, 2], [256, 512, 512]]  
YOLO11m-pose summary: 246 layers, 20,903,589 parameters, 20,903,573 gradients, 71.9 GFLOPs

Transferred 679/691 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
Downloading https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt to 'yolo11n.pt'...
100%|██████████| 5.35M/5.35M [00:00<00:00, 278MB/s]
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.3±0.1 ms, read: 0.7±0.8 MB/s, size: 737.8 KB)
train: Scanning /content/drive/MyDrive/YOLO_Pose/labels/train.cache... 974 images, 67 backgrounds, 0 corrupt: 100%|██████████| 1041/1041 [00:00<?, ?it/s]WARNING ⚠️ No 'flip_idx' array defined in data.yaml, setting augmentation 'fliplr=0.0'

albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
AutoBatch: Computing optimal batch size for imgsz=640 at 65.0% CUDA memory utilization.
AutoBatch: CUDA:0 (NVIDIA L4) 22.16G total, 0.19G reserved, 0.19G allocated, 21.78G free
      Params      GFLOPs  GPU_mem (GB)  forward (ms) backward (ms)                   input                  output
    20903589       71.89         1.741          60.3           nan        (1, 3, 640, 640)                    list
    20903589       143.8         3.590         49.47           nan        (2, 3, 640, 640)                    list
    20903589       287.6         6.023         51.75           nan        (4, 3, 640, 640)                    list
    20903589       575.1        10.930         67.35           nan        (8, 3, 640, 640)                    list
    20903589        1150        20.261         121.5           nan       (16, 3, 640, 640)                    list
    20903589        2300        39.034         272.1           nan       (32, 3, 640, 640)                    list
CUDA out of memory. Tried to allocate 200.00 MiB. GPU 0 has a total capacity of 22.16 GiB of which 111.38 MiB is free. Process 5860 has 22.05 GiB memory in use. Of the allocated memory 21.38 GiB is allocated by PyTorch, and 423.19 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
AutoBatch: Using batch-size 10 for CUDA:0 13.51G/22.16G (61%) ✅
train: Fast image access ✅ (ping: 0.3±0.1 ms, read: 0.5±0.4 MB/s, size: 443.8 KB)
train: Scanning /content/drive/MyDrive/YOLO_Pose/labels/train.cache... 974 images, 67 backgrounds, 0 corrupt: 100%|██████████| 1041/1041 [00:00<?, ?it/s]albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

val: Fast image access ✅ (ping: 0.4±0.2 ms, read: 0.4±0.2 MB/s, size: 442.9 KB)
val: Scanning /content/drive/MyDrive/YOLO_Pose/labels/val... 91 images, 5 backgrounds, 0 corrupt: 100%|██████████| 96/96 [00:22<00:00,  4.33it/s]val: New cache created: /content/drive/MyDrive/YOLO_Pose/labels/val.cache

Plotting labels to runs/pose/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 112 weight(decay=0.0), 122 weight(decay=0.00046875), 121 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 6 dataloader workers
Logging results to runs/pose/train
Starting training for 600 epochs...

      Epoch    GPU_mem   box_loss  pose_loss  kobj_loss   cls_loss   dfl_loss  Instances       Size
  0%|          | 0/105 [00:00<?, ?it/s]
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
[<ipython-input-5-1a9c6a51a24c>](https://localhost:8080/#) in <cell line: 0>()
----> 1 results = model.train(data='/content/drive/MyDrive/YOLO_Pose/poses.yaml', epochs=600, imgsz= 640, batch=0.65, workers=6, multi_scale=True, cos_lr=True, dfl=1.5, cls=0.75, box=7.75, degrees=90.0, shear=90.0, scale=0.9, mixup=0.15, copy_paste=0.4, flipud=0.25, fliplr=0.25, patience=200)

8 frames
[/usr/local/lib/python3.11/dist-packages/torch/_utils.py](https://localhost:8080/#) in reraise(self)
    731             # instantiate since we don't know how to
    732             raise RuntimeError(msg) from None
--> 733         raise exception
    734 
    735 

error: Caught error in DataLoader worker process 0.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.11/dist-packages/torch/utils/data/_utils/worker.py", line 349, in _worker_loop
    data = fetcher.fetch(index)  # type: ignore[possibly-undefined]
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/torch/utils/data/_utils/fetch.py", line 52, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/torch/utils/data/_utils/fetch.py", line 52, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "/usr/local/lib/python3.11/dist-packages/ultralytics/data/base.py", line 369, in __getitem__
    return self.transforms(self.get_image_and_label(index))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/ultralytics/data/base.py", line 383, in get_image_and_label
    label["img"], label["ori_shape"], label["resized_shape"] = self.load_image(index)
                                                               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dist-packages/ultralytics/data/base.py", line 237, in load_image
    im = cv2.resize(im, (w, h), interpolation=cv2.INTER_LINEAR)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cv2.error: OpenCV(4.11.0) /io/opencv/modules/imgproc/src/resize.cpp:3845: error: (-215:Assertion failed) !dsize.empty() in function 'resize'

--- a/ultralytics/__init__.py
+++ b/ultralytics/__init__.py
@@ -1,6 +1,6 @@
 # Ultralytics 🚀 AGPL-3.0 License - https://ultralytics.com/license
 
-__version__ = "8.3.114"
+__version__ = "8.3.115"
 
 import os
 

--- a/ultralytics/utils/patches.py
+++ b/ultralytics/utils/patches.py
@@ -31,7 +31,8 @@ def imread(filename: str, flags: int = cv2.IMREAD_COLOR):
     if filename.endswith((".tiff", ".tif")):
         success, frames = cv2.imdecodemulti(file_bytes, cv2.IMREAD_UNCHANGED)
         if success:
-            return np.stack(frames, axis=2)  # or np.asarray(frames).transpose(1,2,0)
+            # handle RGB images in tif/tiff format
+            return frames[0] if len(frames) == 1 and frames[0].ndim == 3 else np.stack(frames, axis=2)
         return None
     else:
         return cv2.imdecode(file_bytes, flags)

